packages/ec-core/src/test_results.rs

```diff
@@ -349,4 +338,59 @@ mod test_results_from_vec {
         assert_eq!(test_results.results, results.collect::<Vec<_>>());
         assert_eq!(test_results.total_result, scores.into_iter().sum());
     }
+
+    #[test]
+    fn can_clone_test_results_of_scores_of_i32() {
+        let scores = vec![5, 8, 0, 9];
+        let test_results: TestResults<Error<i32>> = scores.into();
+        #[expect(clippy::redundant_clone, reason = "We're testing Clone")]
+        let cloned_test_results = test_results.clone();
+        assert_eq!(cloned_test_results, test_results);
+    }
+
+    #[test]
+    fn can_clone_test_results_of_errors_of_i32() {
+        let errors = vec![5, 8, 0, 9];
+        let test_results: TestResults<Error<i32>> = errors.into();
+        #[expect(clippy::redundant_clone, reason = "We're testing Clone")]
+        let cloned_test_results = test_results.clone();
+        assert_eq!(cloned_test_results, test_results);
+    }
+
+    // The following code is commented out because it doesn't compile. I wrote
+    // it to demonstrate to myself that we indeed could not call `.clone()`
+    // on a `TestResults` of a type that doesn't implement `Clone`, and to
+    // see what the exact error was. It was pretty straightforward, saying
+    // that `NonCloneable` doesn't implement `Clone`.
+
+    // struct NonCloneable(i32);
+    //
+    // impl From<NonCloneable> for i32 {
+    //     fn from(NonCloneable(i): NonCloneable) -> Self {
+    //         i
+    //     }
+    // }
+    //
+    // impl<'a> Sum<&'a NonCloneable> for NonCloneable {
+    //     fn sum<I>(iter: I) -> Self
+    //     where
+    //         I: Iterator<Item = &'a NonCloneable>,
+    //     {
+    //         NonCloneable(iter.into_iter().map(|NonCloneable(i)| i).sum())
+    //     }
+    // }
+    //
+    // #[test]
+    // fn cannot_clone_test_results_of_scores_of_non_cloneable() {
+    //     let scores: Vec<NonCloneable> = vec![
+    //         NonCloneable(5),
+    //         NonCloneable(8),
+    //         NonCloneable(0),
+    //         NonCloneable(9),
+    //     ];
+    //     let test_results: TestResults<NonCloneable> = scores.into();
+    //     #[expect(clippy::redundant_clone, reason = "We're testing Clone")]
+    //     let cloned_test_results = test_results.clone();
+    //     assert_eq!(cloned_test_results, test_results);
+    // }
```

Comment:
Again, see the above comment. If we want to test this (huge if) we could use `static_assertions::assert_not_impl_any!()` instead.

Reply:
But honestly this just seems to me more like you playing around with the derive macro to see how it works, which is fine I guess if it helped you understand them better, but I wouldn't actually put it in the project and just delete it

Reply:
And probably this has even less value than the above assert.